curriculum/challenges/english/25-front-end-development/lab-book-organizer/67172b43f84bcd2dec238a3d.md

```diff
@@ -0,0 +1,164 @@
+---
+id: 67172b43f84bcd2dec238a3d
+title: Build a Book Organizer
+challengeType: 14
+dashedName: build-a-book-organizer
+---
+
+# --description--
+
+**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab. 
+
+**User Stories:**
+
+1. You should have an array of objects named `books` where each object in the array should have a string `title`, another string `authorName`, and a number `releaseYear`. 
+
+2. Your `books` array should have a minimum of three objects.
+
+3. You should have a callback function named `sortByYear` that accepts two books, `book1`, and `book2` as parameter for sorting the array.
+
+4. The `sortByYear` function should return `-1` if the `releaseYear` of `book1` object is smaller than that of the `book2` object.
+
+5. The `sortByYear` function should return `1` if the `releaseYear` of `book1` object is bigger than that of the `book2` object.
+
+6. The `sortByYear` function should return `0` for all other scenarios.
+
+7. You should call the `filter` higher-order function on the `books` array with a callback to filter out books written after a certain year such as 1950 and save the filtered array in a new array named `filteredBooks`.
+
+8. You should call the `sort` higher order function by passing the `sortByYear` callback function on the `filteredBooks` array to sort the books according to their `releaseYear` in ascending order and save the sorted array in a new array named `filteredBooksSorted`.
+
+9. You should call the `forEach` higher order function on the `filteredBooksSorted` array and print out the details of each book following the `<BOOK_TITLE> by <BOOK_AUTHOR> (<releaseYear>)` format.
```

Comment:
After careful consideration, these 3 user stories and their corresponding hints and tests should be more open ended. It will go more with the spirit of the labs and match better with the other labs.

We already have the workshop that comes immediately before this were the tests are much stricter to look for `filter,` `forEach`, etc. 

In this lab, it should be more open ended and not dependent on regex tests. We will just test the final outcome with other assert methods. 

The wording for all of these should be updated to just check that they have create the variable and assigned it the result of a filtered array, or sorted array, etc.

This will mean that some campers might use other methods for solving this, but I actually think that is ok because like I mentioned earlier, we do have the workshop that will be testing for specific methods 👍

Reply:
I have updated the stories to leave out any concrete solutions and rely on the camper's discretion.